### PR TITLE
For #12016: Follow ux mock on Settings --> Data Collection

### PR DESCRIPTION
**Pull request by hakkikaancaliskan**
resim

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [x] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by hakkikaancaliskan** [comment]
Cc @brampitoyo this one ready for review.

File: app/src/main/res/xml/data_choices_preferences.xml [hunk 1]
```diff
@@ -4,19 +4,22 @@
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto">
-    <androidx.preference.SwitchPreference
+    <SwitchPreference
         android:key="@string/pref_key_telemetry"
         android:summary="@string/preferences_usage_data_description"
         android:title="@string/preference_usage_data"
+        android:layout="@layout/checkbox_right_preference"
         app:iconSpaceReserved="false" />
```
Comment (NotWoods) [inline app/src/main/res/xml/data_choices_preferences.xml:12]: Setting `app:iconSpaceReserved="true"` (or removing it as true is the default) seems to set a 72dp margin, so a custom layout isn't needed.

Comment (hakkikaancaliskan) [reply app/src/main/res/xml/data_choices_preferences.xml:12]: Nice :)
(currently swearing to himself)

**Comment by brampitoyo** [comment]
@hakkikaancaliskan Looks good to me!

**Comment by ekager** [comment]
Closing for https://github.com/mozilla-mobile/fenix/pull/12325

**Pull request closed without merging**